File: frontend/src/api/tts.ts
import { buildApiUrl, parseJsonResponse } from "./client";

export type TtsSynthesisRecord = {
  job_id: number;
  audio_file_path: string | null;
  audio_file_url: string | null;
  engine_used: string;
  status: string;
  error: string | null;
};

export async function synthesizeSpeech(input: {
  text: string;
  voice_profile?: string;
  engine?: string;
}) {
  const response = await fetch(await buildApiUrl("/api/tts/synthesize"), {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(input),
  });

  return parseJsonResponse<TtsSynthesisRecord>(response);
}

File: backend/app/schemas/tts.py
from pydantic import BaseModel, Field


class TtsSynthesisRequest(BaseModel):
    text: str = Field(min_length=1)
    voice_profile: str | None = Field(default=None, max_length=100)
    engine: str | None = Field(default=None, max_length=100)


class TtsSynthesisResponse(BaseModel):
    job_id: int
    audio_file_path: str
    audio_file_url: str | None = None
    engine_used: str
    status: str
